src/Microsoft.DotNet.Build.Tasks/GetVersionFromAssembly.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GetVersionFromAssembly : Task
+    {
+        [Required]
+        public string PathToAssembly { get; set; }
+
+        [Output]
+        public string AssemblyVersion { get; set; }
+
+        public override bool Execute()
+        {
+            if (!File.Exists(PathToAssembly))
+            {
+                Log.LogError($"{PathToAssembly} does not exist.");
+                return false;
+            }
+            try
+            {
+                Regex regex = new Regex(@"Commit Hash: ([A-Za-z0-9].*)");
+                string productVersion = FileVersionInfo.GetVersionInfo(PathToAssembly).ProductVersion;
+                Match match = regex.Match(productVersion);
+                if (match.Success && match.Groups.Count > 1)
```

Comment:
@ericstj should we go ahead and have this task expose the other version information from this assembly?